### ISSUE TITLE
Error while reading file “.bashrc”

### ISSUE THREAD

**Issue by yuceltoluyag** [fish-shell/fish-shell#5559]
```
fish --version
fish, version 3.0.0
```
```
uname -a 

Linux archPC 4.20.3-arch1-1-ARCH #1 SMP PREEMPT Wed Jan 16 22:38:58 UTC 2019 x86_64 GNU/Linux
```

```
echo $TERM
st-256color
```

```
sh -c 'env HOME=$(mktemp -d) fish'
Welcome to fish, the friendly interactive shell
```
```
.bashrc (line 5): $(...) is not supported. In fish, please use '(tput)'.
export PS1="\[$(tput bold)\]\[$(tput setaf 1)\][\[$(tput setaf 3)\]\u\[$(tput setaf 2)\]@\[$(tput setaf 4)\]\h \[$(tput setaf 5)\]\W\[$(tput setaf 1)\]]\[$(tput setaf 7)\]\\$ \[$(tput sgr0)\]"
              ^
from sourcing file .bashrc
        called on standard input

source: Error while reading file “.bashrc”
```
my .bashrc

```
#!/bin/bash
stty -ixon # Disable ctrl-s and ctrl-q.
shopt -s autocd #Allows you to cd into directory merely by typing the directory name.
HISTSIZE= HISTFILESIZE= # Infinite history.
export PS1="\[$(tput bold)\]\[$(tput setaf 1)\][\[$(tput setaf 3)\]\u\[$(tput setaf 2)\]@\[$(tput setaf 4)\]\h \[$(tput setaf 5)\]\W\[$(tput setaf 1)\]]\[$(tput setaf 7)\]\\$ \[$(tput sgr0)\]"

[ -f "$HOME/.shortcuts" ] && source "$HOME/.shortcuts" # Load shortcut aliases

# System Maintainence
alias mw="~/.config/mutt/mutt-wizard.sh"
alias sdn="sudo shutdown now"
alias psref="gpg-connect-agent RELOADAGENT /bye" # Refresh gpg
alias gua="git remote | xargs -L1 git push --all"

# Some aliases
alias e="$EDITOR"
alias p="sudo pacman"
alias SS="sudo systemctl"
alias v="vim"
alias f="vifm"
alias r="ranger"
alias sr="sudo ranger"
alias ka="killall"
alias g="git"
alias trem="transmission-remote"
alias mkd="mkdir -pv"
alias ref="shortcuts >/dev/null ; source ~/.bashrc" # Refresh shortcuts manually and reload bashrc
alias mpv="mpv --input-ipc-server=/tmp/mpvsoc$(date +%s)"
alias x="sxiv -ft *"
alias lsp="pacman -Qett --color=always | less"

# Adding color
alias ls='ls -hN --color=auto --group-directories-first'
alias grep="grep --color=auto"
alias diff="diff --color=auto"
alias ccat="highlight --out-format=ansi" # Color cat - print file with syntax highlighting.

# Internet
alias yt="youtube-dl --add-metadata -i" # Download video link
alias yta="yt -x -f bestaudio/best" # Download only audio
alias YT="youtube-viewer"

# Babanın Mekanı
alias guncelle='sudo pacman -Syyu & aurman -Syu'
alias paketara='sudo pacman -Ss'
alias temizle='sudo pacman -Sc $(pacman -Qdtq)'
alias ptemiz='sudo pacman -Rs $(pacman -Qtdq)'
alias hata='systemctl --failed'
alias hatalog='journalctl -p 3 -xb'
alias ytmp3='youtube-dl -ict --yes-playlist --extract-audio --audio-format mp3 --audio-quality 0'

#genel Ayarlar

export PATH=/home/friday13/script/:$PATH
PATH="$HOME/.config/composer/vendor/bin:$PATH"
PATH="$HOME/.node_modules/bin:$PATH"
export npm_config_prefix=~/.node_modules

shdl() { curl -O $(curl -s http://sci-hub.tw/"$@" | grep location.href | grep -o http.*pdf) ;}
se() { du -a ~/.scripts/* ~/.config/* | awk '{print $2}' | fzf | xargs  -r $EDITOR ;}
sv() { vcopy "$(du -a ~/.scripts/* ~/.config/* | awk '{print $2}' | fzf)" ;}
vf() { fzf | xargs -r -I % $EDITOR % ;}

```

**Comment by ridiculousfish** [comment]
fish should not attempt to read .bashrc because it has incompatible syntax. Are you sourcing .bashrc directly, or is something doing this automatically?

**Comment by faho** [comment]
To provide some additional information:

```bash
stty -ixon # Disable ctrl-s and ctrl-q.
shopt -s autocd #Allows you to cd into directory merely by typing the directory name.
HISTSIZE= HISTFILESIZE= # Infinite history.
```

Fish does all of these already, without any configuration.

```bash
export PS1="\[$(tput bold)\]\[$(tput setaf 1)\][\[$(tput setaf 3)\]\u\[$(tput setaf 2)\]@\[$(tput setaf 4)\]\h \[$(tput setaf 5)\]\W\[$(tput setaf 1)\]]\[$(tput setaf 7)\]\\$ \[$(tput sgr0)\]"
```

In fish, the prompt is defined in the `fish_prompt` _function_. So instead of putting together a cryptic string in one variable with escapes that are recognized only here, you'd write some code.

Or you'd execute `fish_config` and pick one of our sample prompts.

We also provide `set_color` as a replacement for `tput`. Instead of numbers, it takes color names or RGB values. So `tput setaf 1` would be `set_color red`.

```bash
[ -f "$HOME/.shortcuts" ] && source "$HOME/.shortcuts" # Load shortcut aliases
```

Now, this syntax is available in fish 3.0, but the source'd file is probably a bashscript which can have syntax that isn't.

Your aliases work for the most part, except fish uses `(command)` instead of `$(command)`.

So:

```bash
alias mpv="mpv --input-ipc-server=/tmp/mpvsoc$(date +%s)"
```

would be

```fish
alias mpv="mpv --input-ipc-server=/tmp/mpvsoc(date +%s)"
```

(though note that in the bash version, `date +%s` is executed when .bashrc is sourced! Run `mpv` twice and it'll use the same file. That's _probably_ not what you want)

```bash
export PATH=/home/friday13/script/:$PATH
```

This is _almost_ available in fish 3, you just need to double-quote `$PATH`: `export PATH=/home/friday13/script/:"$PATH"`.

However, the more usual syntax is:

```fish
set --prepend PATH /home/friday13/script
```

as $PATH is a list.

```bash
PATH="$HOME/.config/composer/vendor/bin:$PATH"
```

This would _have_ to use `set` - `set PATH "$HOME/.config/composer/vendor/bin:$PATH"`.

```bash
shdl() { curl -O $(curl -s http://sci-hub.tw/"$@" | grep location.href | grep -o http.*pdf) ;}
```

fish's function syntax is different, and the various `$@` and `$*` variables don't exist. Function parameters are stored in $argv:

```fish
function shdl
    curl -O (curl -s http://sci-hub.tw/"$argv" | grep location.href | grep -o "http.*pdf")
end
```

(also that `http.*pdf` looks like a glob! If you execute this function in bash in a directory with a file called "http.something.pdf" you'll have surprising results. If bash can't match a glob, it'll pass the expression through unchanged. That's ridiculous, so fish doesn't do it)

Hope that helps!
